src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java

```diff
@@ -140,13 +143,14 @@ public long sendGroupMessage(final String message, final List<String> attachment
         }
     }
 
+    // Since contact names might be empty if not defined, also potentially return
+    // the profile name
+    // Profile names separate firstname/lastname by a \0 - replace by space
+    // otherwise the interface will hickup
     @Override
     public String getContactName(final String number) {
-        try {
-            return m.getContactName(number);
-        } catch (InvalidNumberException e) {
-            throw new Error.InvalidNumber(e.getMessage());
-        }
+        String name = m.getContactOrProfileName(number).replace("\0", " ");
```

Comment:
Interesting, but this shouldn't be in the dbus layer. I've added a `getDisplayName()` method to the profile class, so you can remove the replacing here.